wannierberri/__sym_wann.py

```diff
@@ -0,0 +1,437 @@
+                for atom in range(self.num_wann_atom):
+                    p_map[atom],p_map_dagger[atom] = self.full_p_mat(atom,rot)
+                R_map = np.dot(R_list,np.transpose(self.symmetry['rotations'][rot]))
+                atom_R_map = R_map[:,None,None,:] - vec_shift[None,:,None,:] + vec_shift[None,None,:,:]
+                Ham_all = np.zeros((nRvec,self.num_wann_atom,self.num_wann_atom,self.num_wann,self.num_wann),dtype=complex)
+                matrix_list_all = {X:np.zeros((nRvec,self.num_wann_atom,self.num_wann_atom,self.num_wann,self.num_wann,3),dtype=complex) 
+                            for X in self.matrix_list}
+                
+
+
+                #TODO try numba
+                for iR in range(nRvec):
+                    for atom_a in range(self.num_wann_atom):
+                        for atom_b in range(self.num_wann_atom):
+                            new_Rvec=list(atom_R_map[iR,atom_a,atom_b])
+                            if new_Rvec in self.iRvec:
+                                new_Rvec_index = self.iRvec.index(new_Rvec)
+                                Ham_all[iR,atom_a,atom_b,self.H_select[atom_a,atom_b]] = self.Ham_R[self.H_select[rot_map[atom_a],
+                                    rot_map[atom_b]],new_Rvec_index]
+                                for X in self.matrix_list: 
+                                    if X in ['AA','SS']:
+                                        num_w_a = len(sum(self.wann_atom_info[atom_a][4], [])) #number of orbitals of atom_a
+                                        #X_L: only rotation wannier centres from L to L' before rotating orbitals.
+                                        XX_L = self.matrix_list[X][self.H_select[rot_map[atom_a],rot_map[atom_b]],new_Rvec_index,:].reshape(num_w_a,num_w_a,3)
+                                        #special even with R == [0,0,0] diagonal terms.
+                                        if iR == self.iRvec.index([0,0,0]) and atom_a == atom_b:
+                                            if X == 'AA':
+                                                XX_L += np.einsum( 'mn,p->mnp',np.eye(num_w_a),(vec_shift[atom_a]-self.symmetry['translations'][rot]).dot(self.lattice))
+                                        #X_all: rotating vector.
+                                        matrix_list_all[X][iR,atom_a,atom_b,self.H_select[atom_a,atom_b],:] = np.einsum('ij,nmi->nmj',self.rot_c[rot],XX_L).reshape(-1,3)
+                                    else: 
+                                        print (f"WARNING: I do not know how to symmetrize  {X}")
+                            else:
+                                norm = np.linalg.norm(self.Ham_R[self.H_select[rot_map[atom_a],rot_map[atom_b]],iR])
+                                if norm > 1e-8:
+                                    print (f"WARNING : after symmetry transformation got a new Rvec: {new_Rvec} for iR = {iR}, Rvec = {self.iRvec[iR]} and atoms {atom_a} and {atom_b}"+
```

Comment:
This warning does not happen if `use_ews = True` is used. We could force use_ws before symmetrization, but `System_tb` does not have `mp_grid` if it is not provided manually